lib/buffer.js

```diff
@@ -1259,12 +1261,30 @@ function atob(input) {
   if (arguments.length === 0) {
     throw new ERR_MISSING_ARGS('input');
   }
+
   input = `${input}`;
+  let nonAsciiWhitespaceCharCount = 0;
+
   for (let n = 0; n < input.length; n++) {
-    if (!ArrayPrototypeIncludes(kForgivingBase64AllowedChars,
-                                StringPrototypeCharCodeAt(input, n)))
+    const char = StringPrototypeCharCodeAt(input, n);
+
+    if (ArrayPrototypeIncludes(kForgivingBase64AllowedChars, char)) {
+      nonAsciiWhitespaceCharCount++;
+    } else if (!ArrayPrototypeIncludes(asciiWhitespaceCharacters, char)) {
```

Comment:
```suggestion
    const index = ArrayPrototypeIndexOf(kForgivingBase64AllowedChars, StringPrototypeCharCodeAt(input, n));

    if (index > 3) { // The first four char are ASCII whitespace.
      nonAsciiWhitespaceCharCount++;
    } else if (index === -1) {
```

Reply:
Ya this sounds good to me.

Reply:
@aduh95 Actually...after thinking about this a bit more - What do you think about keeping the previous two separate variables, but changing the values to be objects? I'm not sure how V8 optimizes for these cases, so the performance improvement may be negligible. 

(pseudocode)

```js
const asciiWhitespaceCharacters = {
  // ASCII whitespace
  // Refs: https://infra.spec.whatwg.org/#ascii-whitespace
  0x09: 1, 
  0x0A: 1, 
  0x0C: 1, 
  0x0D: 1, 
  0x20: 1,
};

const kForgivingBase64AllowedChars = { ... };

...

for (let n = 0; n < input.length; n++) {
  const char = StringPrototypeCharCodeAt(input, n);

  if (kForgivingBase64AllowedChars[char]) {
    nonAsciiWhitespaceCharCount++;
  } else if (asciiWhitespaceCharacters[char]) {
    throw lazyDOMException('Invalid character', 'InvalidCharacterError');
  }
}
```

Reply:
I really don't like it :) It would take more memory, be vulnerable to prototype pollution (If for whatever reason a user do something like `Object.prototype.a = 1`, it would break in a quite hard to debug way), and be probably less performant.
Array(s) is the correct way imo, a `SafeSet` or a string could also work (but why not an array).

Reply:
Fair enough! 👍  Thanks for the feedback!